### ISSUE TITLE
MinLen and Varargs

### ISSUE THREAD

**Issue by mernst** [kelloggm/checker-framework#136]
The Index Checker issues a warning for the first invocation of `check`, but not the second one, in the following code:

```java
import org.checkerframework.checker.index.qual.MinLen;

class MinLenVarargs {
  static void check(String @MinLen(2) ... var) {
    System.out.println(var[0] + " " + var[1]);
  }

  public static void main(String[] args) {
    //:: error: (argument.type.incompatible)
    check(new String[] {"goodbye"});
    //:: error: (argument.type.incompatible)
    check("goodbye");
  }
}
```

**Comment by kelloggm** [comment]
Here's an updated test case that includes all the possible cases of varargs for MinLen:

```java
import org.checkerframework.checker.index.qual.MinLen;

class MinLenVarargs {
  static void check(String @MinLen(3) ... var) {
    System.out.println(var[0] + " " + var[1]);
  }

  public static void main(String[] args) {
    //:: error: (argument.type.incompatible)
    check(new String[] {"goodbye"});
    //:: error: (argument.type.incompatible)
    check();
    //:: error: (argument.type.incompatible)
    check("goodbye");
    //:: error: (argument.type.incompatible)
    check("hello", "goodbye");
  }
}
```

There are three interesting varargs cases, and the MLC only gets one of them correct right now (the first one in the test case, and also in the following list). Here are the cases:
* When an array that typechecks is passed, it is used as the value. The call `check(new String[] {"goodbye"});` above is an example, which does issue an error.
* When nothing is passed, then an empty array is created and passed. The call to `check()` above, which does not produce an error, is an example.
* When objects that typecheck as elements of the array (one or more) are passed, an array is created and the elements are put into the array (the two last calls to `check` in the code above,
neither of which produce errors, are examples).

**Comment by kelloggm** [comment]
Fundamentally, it seems like the way that the CFG builder handles all of these cases is to create an `ArrayCreationNode` if a new array is required, but not to associate a tree with it (i.e. null is passed). The result is that `MinLenAnnotatedTypeFactory#MinLenTreeAnnotator` can't type the artificial array.

**Comment by kelloggm** [comment]
Closed in favor of typetools#1218. This is a general issue that applies to more than just the Index Checker.
